File: NavitasBeta/NavitasBeta/PageParameterList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Collections.ObjectModel;

namespace NavitasBeta
{
    public class PageParameterList
    {
        public NavitasGeneralPage pagePointer;
        public bool Active = true;
        public long uniqueID = 0;
        public int PacketsEvaluatedForSending = 0;
        public string ParentTitle = "";
        public enum ParameterListType : int { TAC, TSX };
        public ParameterListType parameterListType;
        public List<GoiParameter> parameterList = new List<GoiParameter>();

        public PageParameterList(ParameterListType pageType, NavitasGeneralPage parentPage)
        {
            parameterListType = pageType;
            uniqueID = DateTime.Now.Ticks; //good enough for running a year instead of GUID

            ParentTitle = parentPage.Title;
            pagePointer = parentPage;
        }
        public PageParameterList() { } //somehow makes exposes it of typeof() so that I can serialize it?

    }
}

File: NavitasBeta/NavitasBeta/CommonWebViews/frameworks/PageParameterList.js
class PageParameterList extends List {
    static ParameterListType = { TAC: 0, TSX: 1 };
    static ProtocalCommandTypes = { READ:0x20, WRITE: 0x21}; //protocal actually does something different but you get the idea.
    //fields declared here to be public
    Active = null;
    ProtocalCommandType = "";
    uniqueID = null;
    ParentTitle = "";
    parameterListType = null;
    parameterList = [];
    timeCreated = null;
    constructor(pageType, parentTitle) {
        super();
        var _this = this; //_this allows this to be exposed to constructor functions
        //fields declared here are private?
        this.Active = false;
        this.uniqueID = Math.random()*Date.now();
        this.ParentTitle = parentTitle;
        this.parameterListType = pageType;
        this.parameterList = new List();
        this.timeCreated = Date.now();
        //public PageParameterList() { } //somehow makes exposes it of typeof() so that I can serialize it?
    }
}
